modules/renter/upload.go

```diff
@@ -152,13 +154,40 @@ func (r *Renter) Upload(up modules.FileUploadParams) error {
 		UploadParams:   up,
 		renter:         r,
 	}
+	r.save()
+
+	// Upload a piece to every host on the network.
+	errChan := make(chan error, len(r.files[up.Nickname].Pieces))
 	for i := range r.files[up.Nickname].Pieces {
-		// threadedUploadPiece will change the memory that the piece points to,
-		// which is useful because it means the file itself can be renamed but
-		// will still point to the same underlying pieces.
-		go r.threadedUploadPiece(up, &r.files[up.Nickname].Pieces[i])
+		// threadedUploadPiece will change the memory that the piece points
+		// to, which is useful because it means the file can be renamed
+		// without disrupting the upload process.
+		go func(piece *filePiece) {
+			errChan <- r.threadedUploadPiece(up, piece)
+		}(&r.files[up.Nickname].Pieces[i])
+	}
+	reqPieces := r.files[up.Nickname].PiecesRequired
+	r.mu.Unlock(lockID)
```

Comment:
can't do separate scoping because it would capture local variables. (i.e. `reqPieces` and `errChan` would be undefined)